[New issue] CMake issue due to ioda dependency

## Description

Recently in testing a fix on `JCSDA-internal/ufo` (see https://github.com/JCSDA-internal/ufo/issues/3793), I encountered a build issue with `ufo-data`. 

In ufo, a recent commit onto ufo `develop`, https://github.com/JCSDA-internal/ufo/commit/3d802418ecdcc2beb3bd88bcc3b2aac1842f0e42 added a:
```cmake
  find_package( ufo-data QUIET )
```

Previously I "skipped" the need for `ufo-data` via the `ENABLE_UFO_DATA` that https://github.com/JCSDA-internal/ufo/commit/3d802418ecdcc2beb3bd88bcc3b2aac1842f0e42 removed. 

So, I then added the `ufo-data` repo and when I tried to build I got:

```
-- [ufo-data] (1.10.0) [ed8ca2e]
-- Feature TESTS enabled
CMake Error at /gpfsm/dulocal15/sles15/other/cmake/3.28.2/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenMP (missing: OpenMP_Fortran_FOUND Fortran) (found
  version "5.1")
Call Stack (most recent call first):
  /gpfsm/dulocal15/sles15/other/cmake/3.28.2/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /gpfsm/dulocal15/sles15/other/cmake/3.28.2/share/cmake-3.28/Modules/FindOpenMP.cmake:633 (find_package_handle_standard_args)
  /gpfsm/dulocal15/sles15/other/cmake/3.28.2/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /discover/swdev/mathomp4/Models/jedi_gaan/install/lib64/cmake/oops/oops-import.cmake:27 (find_dependency)
  /discover/swdev/mathomp4/Models/jedi_gaan/install/lib64/cmake/oops/oops-config.cmake:83 (include)
  /gpfsm/dulocal15/sles15/other/cmake/3.28.2/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /discover/swdev/mathomp4/Models/jedi_gaan/install/lib64/cmake/ioda/ioda-import.cmake:40 (find_dependency)
  /discover/swdev/mathomp4/Models/jedi_gaan/install/lib64/cmake/ioda/ioda-config.cmake:64 (include)
  CMakeLists.txt:21 (find_package)


-- Configuring incomplete, errors occurred!
make[2]: *** [src/CMakeFiles/ufo-data.dir/build.make:91: src/ufo-data-prefix/src/ufo-data-stamp/ufo-data-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:1287: src/CMakeFiles/ufo-data.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

My guess is the issue is that `ufo-data` has a `find_package(ioda)` and `ioda` is a Fortran code. Thus, when `ufo-data` pulls in `ioda` it pulls in a dependency on `OpenMP::OpenMP_Fortran`. But since `ufo-data` is declared as:
```cmake
project( ufo-data VERSION 1.10.0 DESCRIPTION "UFO Test Files" )
```
In CMake a "bare" `project()` is a C and C++ project but not Fortran. I was able to get past this error by setting:
```cmake
project( ufo-data VERSION 1.10.0 DESCRIPTION "UFO Test Files" LANGUAGES C CXX Fortran)
```



## Requirements

Expected behavior is that CMake succeeds.

## Acceptance Criteria (Definition of Done)

The build succeeds without issue.

NOTE: As shown in PRs like https://github.com/JCSDA-internal/ufo/pull/3735 and others in JCSDA land, I am building JEDI code in a different way than "stock". I'm not using `ecbuild_bundle()` but rather more pure CMake (ExternalProject_Add). Thus, it's possible I'm seeing differences in CMake resolution? Not sure.

## Dependencies

This mainly blocks progress I'm having in my non-standard use of JEDI repos. I still have things like https://github.com/JCSDA-internal/ufo/pull/3735 that need fixed before all is "well" for me.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,48 +15,5 @@ include( ecbuild_system NO_POLICY_SCOPE )
 ecbuild_declare_project()
 set( CMAKE_DIRECTORY_LABELS ${PROJECT_NAME} )
 
-# This CMake script adds many tests to validate all of the testing data files
-# within the ufo-data repository.
-
-find_package(ioda REQUIRED)
-
-file( GLOB_RECURSE OBS_FILES *.nc4 *.nc *.ioda )
-# We filter the expression to remove the geovals and obsdiag files, which are geovals-based
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)geoval(.*)" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)obsdiag(.*)" )
-# Remove satbias files
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)satbias(.*)" )
-# Remove misc test files
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)interpolation(.*)" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)obserror_multi_variant(.*)" )
-
-# Remove known buggy non-obs files
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)met_office_gauss_thinning_groundgnssnames_obs.nc4" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)obserror_4d.nc" )  # Wrong dimension attachment
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)obserror_2d_fullr.nc" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)obserror_latitude_longitude.nc" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)satwind_pressure_errors.nc4" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)satwind_pressure_errors_nomissing.nc4" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)scatwind_obs_2d_2020100106.nc4" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)superob_testdata.nc4" )
-list( FILTER OBS_FILES EXCLUDE REGEX "(.*)mtgirs_obserror_fstest.nc4" )  # used with drawFromFile obsfunction
-
-foreach ( f ${OBS_FILES} )
-	get_filename_component(filename ${f} NAME)
-
-	# Note: IODA_YAML_ROOT is provided by find_package(ioda).
-	# The ObsSpace.yaml file is *not* a test file. It always exists.
-	ecbuild_add_test(
-		TARGET ufo_data_validate_${filename}
-		COMMAND ioda-validate.x
-		LABELS ufo_data_validate
-		ENVIRONMENT "ECKIT_COLOUR_OUTPUT=1"
-		ARGS "--ignore-warn"
-                     "--ignore-error"
-                     "${IODA_YAML_ROOT}/validation/ObsSpace.yaml"
-                     "${f}"
-		)
-endforeach()
-
 ecbuild_install_project( NAME ${PROJECT_NAME} )
 ecbuild_print_summary()

